Source/minitext.cpp

```diff
@@ -67,19 +64,77 @@ void InitQuestText()
 	qtextflag = FALSE;
 }
 
+int GetWrap(const char *p)
+{
+	char tempstr[128];
+	int l, i, w;
+	BOOL doneflag;
+	BYTE c;
+
+	int lines = 0;
+
+	doneflag = FALSE;
+	while (!doneflag) {
+		w = 0;
+		const char *s = p;
+		l = 0;
+		while (*s != '\n' && *s != '|' && w < 543) {
+			c = gbFontTransTbl[(BYTE)*s];
+			s++;
+			if (c != '\0') {
+				tempstr[l] = c;
+				w += mfontkern[mfontframe[c]] + 2;
+			} else {
+				l--;
+			}
+			l++;
+		}
+		tempstr[l] = '\0';
+		if (*s == '|') {
+			tempstr[l] = '\0';
+			doneflag = TRUE;
+		} else if (*s != '\n') {
+			while (tempstr[l] != ' ' && l > 0) {
+				tempstr[l] = '\0';
+				l--;
+			}
+		}
+		for (i = 0; tempstr[i]; i++) {
+			p++;
+			if (*p == '\n') {
+				p++;
+			}
+		}
+		lines++;
+	}
+
+	return lines;
+}
+
+void CalcTextSpeed(int n)
+{
+	DWORD SfxFrames, TextHeight;
+
+	SfxFrames = GetSoundFrames(n);

```

Comment:
I cleaned up the audio code and added a helper function so this can now be done the following way:
```suggestion
	SfxFrames = GetSFXLength(n) * 20 / 1000;
```

Reply:
nice! what do you mean btw? isn't it still drawn at 20 FPS? the reason for this is to compensate lag at any given frame rate.
```c
double qtextDelay = (currTime - sgLastScroll) / (1000.0 / 20);
scrolltexty -= (DWORD)(qtextSpd * qtextDelay);
```
This code right here determines what % of the frame we are on and adjusts the speed so if say you drag the window even though the speech is still playing the game stops drawing so when it returns the text will draw faster to stay in sync with the speaker. It took forever to implement this btw, since the PSX used hardware functions.

Reply:
No the mouse and scrolling test is rendered at you monitors refresh rate, for extra smooth goodness.

Reply:
The text seems to still be rendered the same as everything else to me.

Reply:
For any text with a qtextSpd of 0 or higher (txtspd 5 or lower), you won't be able to see a difference as the text won't move more than 1px pr 50 ms. But for some of the faster texts, there should be a perceptual difference.
I think Gillian usually has a speed of -1 so you should be able to start to make it out there when talking to her. Or talk to Farnham about the Poisoned Water quests for maximum effect :D

Reply:
Yeah it makes much more of a difference to the shorter texts.

One of the key points is that the lore speeches they added on the PSX are so long that an integer pixel count will eventually cause a massive de-sync between the text and the speaker. They had to implement decimal pixel count. Vanilla did not have any speeches long enough for this to happen.